### PR TITLE
Use a simpler way to check for symlinks

### PR DESCRIPTION
**Pull request by sschuberth**
That does not involve comparing strings.

### CODE REVIEW

File: clikt/src/jvmMain/kotlin/com/github/ajalt/clikt/parameters/types/file.kt [hunk 1]
```diff
@@ -17,7 +18,7 @@ private fun pathType(fileOkay: Boolean, folderOkay: Boolean): String = when {
 }
 
 private fun File.isSymlink(): Boolean {
-    return toPath().toRealPath().toFile() != absoluteFile
+    return Files.isSymbolicLink(toPath())
```
Comment (ajalt) [inline clikt/src/jvmMain/kotlin/com/github/ajalt/clikt/parameters/types/file.kt]: Lets get rid of the custom `fun Files.isSymlink` entirely and just use `Files.isSymbolicLink` directly.

Comment (ajalt) [reply clikt/src/jvmMain/kotlin/com/github/ajalt/clikt/parameters/types/file.kt]: We'll also want to make this change in `path.kt`, which uses the same logic.

**Comment by sschuberth** [comment]
> We discussed the current implementation in [#114 (comment)](https://github.com/ajalt/clikt/issues/114#issuecomment-580354908), and I was under the impression that you didn't think `Files.isSymbolicLink` works on Windows. Did I misunderstand?

There were indeed some misunderstandings, on both sides. I believe we've clarified that Cygwin's legacy  "symlinks" are not symlinks, so let's leave them aside. For just *detecting* a symlink `Files.isSymbolicLink` indeed works fine on Windows. But if you actually need to resolve the path of a symlink, Java's good old `File.getCanonicalFile` does *not* work, and you need to use something like `toPath().toRealPath().toFile()` instead. Having the latter code, it seemed natural to me to implement a check by comparing paths, but it's in fact nicer to just compare file attributes.

File: clikt/src/jvmMain/kotlin/com/github/ajalt/clikt/parameters/types/file.kt (hunk 1 above)
Comment (sschuberth) [reply clikt/src/jvmMain/kotlin/com/github/ajalt/clikt/parameters/types/file.kt]: Both done.

**Comment by ajalt** [comment]
That makes sense. Thanks!

**Pull request merged**